test/test_opsum_to_ttn.jl

```diff
@@ -125,19 +137,74 @@ using Test
     end
   end
 
+  @testset "OpSum to TTN Fermions" begin
+    # small comb tree
+    auto_fermion_enabled = ITensors.using_auto_fermion()
+    if !auto_fermion_enabled
+      ITensors.enable_auto_fermion()
+    end
+    tooth_lengths = fill(2, 3)
+    c = named_comb_tree(tooth_lengths)
+    is = siteinds("Fermion", c; conserve_nf=true)
+
+    # test with next-nearest neighbor tight-binding model
+    t = 1.0
+    tp = 0.4
+    U = 0.0
+    h = 0.5
+    H = ITensorNetworks.tight_binding(c; t, tp, h)
+
+    # add combination of longer range interactions
+    Hlr = copy(H)
+
+    @testset "Svd approach" for root_vertex in leaf_vertices(is)
+      # get TTN Hamiltonian directly
+      Hsvd = TTN(H, is; root_vertex=root_vertex, cutoff=1e-10)
+      # get corresponding MPO Hamiltonian
+      sites = [only(is[v]) for v in reverse(post_order_dfs_vertices(c, root_vertex))]
+      vmap = Dictionary(reverse(post_order_dfs_vertices(c, root_vertex)), 1:length(sites))
+      Hline = ITensors.MPO(relabel_sites(H, vmap), sites)
+      # compare resulting sparse Hamiltonians
+      Hmat_sp = hopping_hamiltonian(relabel_sites(H, vmap))
+      @disable_warn_order begin
+        Tmpo = prod(Hline)
+        Tttno = contract(Hsvd)
+      end
+
+      # verify that the norm isn't 0 and thus the same (which would indicate a problem with the autofermion system
+      @test norm(Tmpo) > 0
+      @test norm(Tttno) > 0
+      @test norm(Tmpo) ≈ norm(Tttno) rtol = 1e-6
+
+      @test_broken Tmpo ≈ Tttno # ToDo fix comparison for fermionic tensors
+      # In the meantime: matricize tensors and convert to dense Matrix to compare element by element
+      dTmm = to_matrix(Tmpo)
+      dTtm = to_matrix(Tttno)
+      @test any(>(1e-14), dTmm - dTtm)
+
+      # also compare with energies obtained from single-particle Hamiltonian
+      GS_mb, _, _ = eigsolve(dTtm, 1, :SR, eltype(dTtm))
+      spectrum_sp = eigvals(Hmat_sp)
+      @test minimum(cumsum(spectrum_sp)) ≈ GS_mb[1] atol = 1e-8
+    end
+    if !auto_fermion_enabled
+      ITensors.disable_auto_fermion()
+    end
+  end
+
   @testset "OpSum to TTN QN missing" begin
     # small comb tree
     tooth_lengths = fill(2, 3)
     c = named_comb_tree(tooth_lengths)
     c2 = copy(c)
+    ## add an internal vertex into the comb graph c2
     import ITensorNetworks: add_vertex!, rem_vertex!, add_edge!, rem_edge!
```

Comment:
I only use `import` if I am overloading a function, not just bringing it into the namespace to call it. Also it would be better to move this to the top of the file, and bring it into the namespace from the `Graphs.jl` package where they are originally defined, i.e. `using Graphs: add_vertex!, rem_vertex!, add_edge!, rem_edge!`.